### ISSUE TITLE
Please adapt the bootloader to work with different flash-sizes of the ATSAME51 family.

### ISSUE THREAD

**Issue by RudolphRiedel** [jgressmann/supercan#3]
The page size across all E5x/D5x is 512 byte.
But the page size is also available in the NVMCTRL register PARAM as PSZ[2:0].
This supposedly has a value of 0x6 across all members of the family which means 512 byte.
And I just check with the debugger, both the E51J19A and the E51J18A have a 6 in these bits.

Regarding the total flash size, NVMCTRL register PARAM has NVMP[15:0]
which is the amount of pages.
And correctly so the E51J18A has a value of 0x200 in these bits and the E51J19A has a value of 0x400.
So the E51J20A should show 0x800.

**Comment by jgressmann** [comment]
I added support for this feature on the master branch. Since I only have a SAME51J19A board available for testing, I couldn't validate if the code works with other boards. 

You could try the firmware built from the master branch sources to see if it works for you.

**Comment by RudolphRiedel** [comment]
I just tried it with an E51J18A and after the usual trickery with Zadig since I recently re-installed m Win10, it appears to be working just fine, at least on the USB side of things, still need to test the CAN but this is more releated to the board itself.
I read back the flash with Microchip Studio and there is indeed code at 0x20000 or the upper 128k - nice.

**Comment by jgressmann** [comment]
Excellent, please let me know how it goes.

**Comment by RudolphRiedel** [comment]
I flashed all four interfaces I currently have here and two of these have an E51J18A, the other two have E51J19A.
And three of these work just fine.

The fourth one is the one I finished soldering yesterday and which has the E51J18A and alternative digital isolators populated.
So either I messed up on the CAN side or the digital isolators are not working as expected.

Very nice.

Is the new bootloader also supporting the E51J20A?
